File: clients/rn/src/engine/room/RoomEngineResponse.ts
/**
 * Responses from a RoomEngine command.
 * TypeScript port of KMP's RoomEngineResponse.kt.
 */

import type { RoomSnapshot } from '../../protocol/models';

export type RoomEngineResponse = OkResponse | RejectedResponse;

export interface OkResponse {
  type: 'ok';
  snapshot: RoomSnapshot;
}

export interface RejectedResponse {
  type: 'rejected';
  code: string;
  reason: string;
}

File: clients/kmp/shared/src/commonMain/kotlin/org/wyrdsekai/app/engine/room/RoomEngineResponse.kt
package org.wyrdsekai.app.engine.room

import org.wyrdsekai.app.protocol.RoomSnapshot

/** Responses from a RoomEngine command. */
sealed class RoomEngineResponse {
    data class Ok(val snapshot: RoomSnapshot) : RoomEngineResponse()
    data class Rejected(val code: String, val reason: String) : RoomEngineResponse()
}
